[PWA-428] Update upward-connector PWA install docs (#2235)
magento/pwa-studio: pwa-devdocs/src/tutorials/cloud-deploy/index.md

    You are not required to publish to npmjs.com, but
    NPM or Yarn should be able to access your project code through Git.

For this tutorial, the [`@magento/venia-concept`][] package for the [Venia storefront][] is used as a template, but any PWA available though NPM with an UPWARD compatible YAML file is supported.

These instructions provide a method for building your application bundle in the Magento Cloud, but

> you can get the same results by building locally and checking your local build artifacts into source control.

Reviewer: ```suggestion
you can get the same results by building locally with the same environment variables as your targeted production environment, and then checking your local build artifacts into source control.
```

Reviewer: My edit could maybe be phrased better, but I think this is important to note, since environment variables could cause the remote build and the local build to produce different artifacts.

You'd hope they would not, because the local simulated "production" environment should include all the non-secret production variables, you don't want to hardcode any secrets into the bundle.

Author: To my knowledge, the only runtime variables are for UPWARD, so just `NODE_ENV` and `MAGENTO_BACKEND_URL`, all others are baked in. This includes `CHECKOUT_BRAINTREE_TOKEN`, but I always considered that a public key that wasn't useful by itself. Definitely worth auditing if that's not the expectation.

"magento/sample-data-media-venia": "dev-master",
```

## Install dependencies

### Composer dependencies

> _Optional_: If you manually copied the `composer.json` file changes from previous steps, you'll need to update your `composer.lock` with these new dependencies.

Reviewer 2: ```suggestion
_Optional_: If you manually modified the `composer.json` file to make the changes described in the previous steps, update your `composer.lock` with these new dependencies:
```

"magento/sample-data-media-venia": "dev-master",
```

## Install dependencies

### Composer dependencies

_Optional_: If you manually modified the `composer.json` file to make the changes described in the previous steps, update your `composer.lock` with these new dependencies:

```sh
composer update
```

## Specify Cloud server environment

### Install NPX and Yarn

If your project supports Yarn, which is the case for `venia-concept`, add the following entry to the `.magento.app.yaml` file.

```yaml
dependencies:
    nodejs:

>       npx: "*"

Reviewer: Did you find that `npx` actually needs to be installed? On recent versions of Node it comes built-in.

Author: Doesn't seem so, must have got left behind when experimenting with removal of NVM steps. Removed explicit dependency in f934608.

The `venia-concept` storefront uses a newer version of Node than the default environment provides.
It also uses Yarn to install its dependencies.

> The following update to the `hooks.build` section of `.magento.app.yaml` installs NVM and installs the latest LTS release of Node. NPX runs a `buildpack` command  `create-project` that takes arguments to scaffold the application using the `venia-concept` template. Install and build the application, then move that artifact to an area on the file system that is persisted after the build phase.

Reviewer 2: ```suggestion
The following update to the `hooks.build` section of `.magento.app.yaml` installs NVM and the latest LTS release of Node.

It also uses NPX to run the `buildpack` CLI tool with the `create-project` sub-command to scaffold the application using the `venia-concept` template.
After scaffolding completes, it installs and builds the application and moves the build artifact to a persistent area on the file system.
```

        [ -s "$NVM_DIR/nvm.sh"  ] && \. "$NVM_DIR/nvm.sh"
        nvm install --lts=dubnium

>         npx @magento/pwa-buildpack create-project /tmp/pwa --template venia-concept --backend-url https://[your-cloud-url-here] --name Venia --author Magento --install false --npm-client yarn

Reviewer 2: Does this mean developers will have to set up their project so that it can be deployed using the `create-project` tool? Which would mean that they will need to create their own [`create.js`](https://magento.github.io/pwa-studio/pwa-buildpack/scaffolding/custom-project-templates/) instruction file for the command?

Author: I would consider these instructions for people wanting to use `venia-concept` as a turn key solution; but this was a reason I wanted @zetlen to take a peek at these instructions.

It looked like `create-project` is capable of accepting any npm package as a template, but now requires you select one from a list (`venia-concept` being the only option). If we removed that list restriction, anyone who started with a `venia-concept` scaffolded app could publish and use `--template @awesome-si/super-cool-pwa`.

This falls in line with that production deployment story where we've positioned ourselves as unopinionated. We give you instructions for how to produce your bundle, mostly in a local context, but don't get into too much detail on how to make it public.

Reviewer: The `create-project` template functionality is not officially supported yet; there are some kinks to work out, and I haven't notated them yet because I jumped hurriedly off the project to work on BuildBus. So I wouldn't document that quite yet.

Author: Yeah, it looked like the logic was built out far enough to support `create-pwa`, but I could see the long term intention. That should be fine for now, with the title note that local builds are suitable if you're not down for turn key `venia-concept`.
